### DISCUSSION TITLE
Poetry and Makefile

### DISCUSSION THREAD

**Discussion by brunolnetto** [python-poetry/poetry#7739] [General] [answered]
Poetry is built to manage packages, we know that. There is other thing that handle development processes well: **Makefile**! On my learning process, I wrote a [file](https://github.com/trouchet/appy/blob/dbeb741d214624baeab7184a594dee5fa55c4166/Makefile#L88) to organize the main development workflows. 

The commands `make setup` and `make install` works as expected with `sudo`-permission. The command `make enable` outputs the log error below. I do now know why the previous two commands work but this do not.

```
make[1]: *** No rule to make target 'poetry', needed by 'enable'.  Stop.
```

@neersighted It is a great question, isn't it?

**Comment by dimbleby** [comment]
because you got the makefile wrong.

explaining makefiles is very far off topic for a poetry discussion but you want target, then prerequisites, then recipe.  You've written the recipe where the prerequisites should go.

**Comment by brunolnetto** [comment]
The following answer will sound sarcastic, interpret as not: I really want to know what on-topic means in computer science, since most solutions are a combination of tools and practices.

May you inform or forward some reference for me to consult? Or even suggest a fix for given Makefile?

**Reply by dimbleby** [reply]
You are asking this question in the discussion space for the poetry project.  "On-topic" would be discussion about poetry.

https://www.gnu.org/software/make/manual/make.html

it should be very easy to find such things for yourself.

**Comment by brunolnetto** [comment]
Yes, the internet content is some words away and "common knowledge" is way too uncommon than we may think. I thank for your effort to point the proper instructions. As advice, skip the remark "it should be very easy to find such things for yourself."

**Comment by mapsa** [comment, marked as answer]
```
enable: poetry shell
```

A bit too late but your `enable` target is wrong. It is saying `poetry` is a dependency. The Makefile will try to build your dependencies and it is expecting to build your `poetry` target that doesn't exist (it will also try to create a `shell` target after which is also wrong).
It should be:

```
enable:
<tab> poetry shell
```

I have created a simple guidelines for Makefiles here https://github.com/mapsa/makefile-examples in case it helps. I would include phony targets too (`enable` is one).

**Reply by brunolnetto** [reply]
Outstanding. I would read regex explanation as well. :-)
